lib/elixir/src/elixir_erl.erl

```diff
@@ -450,10 +450,8 @@ validate_spec_for_existing_function(ModuleMap, NameAndArity, Line) ->
 % Attributes
 
 attributes_form(Line, Attributes, Forms) ->
-  Fun = fun({Key, Value}, Acc) ->
-    [{attribute, Line, Key, Value} | Acc]
-  end,
-  lists:foldl(Fun, Forms, Attributes).
+  Fun = fun({Key, Value}, Acc) -> [{attribute, Line, Key, Value} | Acc] end,
+  lists:foldr(Fun, Forms, Attributes).
```

Comment:
Why the change to `foldr`?

Reply:
Because persisted attributes were reversed before due to the way we collected them but not anymore.

Reply:
Wouldn't `map` be a better choice in that case?

Reply:
We would change the attributes order in the final erlang abstract format, which I am not sure if it may have consequences or not.

Reply:
```
10> lists:foldr(fun (X, Acc) -> [X * 2 | Acc] end, [], [1, 2, 3]).
[2,4,6]
11> lists:map(fun (X) -> X * 2 end, [1, 2, 3]).
[2,4,6]
```

Reply:
Oh, a `map` operation instead of the data type. It depends. We would need to concatenate the map though to the existing forms.

Reply:
Oh, I completely missed that we have attributes passed already in and it's not an empty list. I'm sorry.